srcpkgs/webkit2gtk/template

```diff
@@ -132,21 +134,132 @@ pre_configure() {
 	esac
 }
 
+post_configure() {
+	mkdir -p webkit2gtk-41
+	pushd webkit2gtk-41
+		configure_args=${configure_args//-DUSE_SOUP2=ON/}
+		configure_args=${configure_args//-DENABLE_WEBDRIVER=OFF/}
+		do_configure
+	popd
+	mkdir -p webkit2gtk-50
+	pushd webkit2gtk-50
+		configure_args+=" -DUSE_GTK4=ON -DENABLE_WEBDRIVER=OFF"
+		do_configure
+	popd
+}
+
+pre_build() {
+	export NINJA_STATUS="[1/3][%f/%t] "
+}
+
+post_build() {
+	export NINJA_STATUS="[2/3][%f/%t] "
+	pushd webkit2gtk-41
+		do_build
+	popd
+	export NINJA_STATUS="[3/3][%f/%t] "
+	pushd webkit2gtk-50
+		do_build
+	popd
+}
+
 post_install() {
+	pushd webkit2gtk-41
+		do_install
+	popd
+	pushd webkit2gtk-50
+		do_install
+	popd
+
 	vlicense Source/WebCore/LICENSE-APPLE
 	vlicense Source/WebCore/LICENSE-LGPL-2.1
 	vlicense Source/WebCore/LICENSE-LGPL-2
 }
 
 webkit2gtk-devel_package() {
+	build_style=meta
+	depends="webkit2gtk40-devel ${sourcepkg}>=${version}_${revision}"
+	short_desc+=" - development files (transitional package)"
+}
+
+webkit2gtk40_package() {
+	depends="${_all_depends}"
+	short_desc="GTK+3 port of the WebKit2 browser engine (soup2)"
+	pkg_install() {
+		vmove usr/lib/webkit2gtk-4.0
+		vmove usr/libexec/webkit2gtk-4.0
+		if [ "$build_option_gir" ]; then
+			vmove "usr/lib/girepository-1.0/*-4.0.typelib"
+		fi
+		vmove "usr/lib/*-4.0.so.*"
+	}
+}
+
+webkit2gtk40-devel_package() {
 	depends="gtk+3-devel libsoup-devel ${sourcepkg}>=${version}_${revision}"
-	short_desc+=" - development files"
+	short_desc="GTK+3 port of the WebKit2 browser engine (soup2) - development files"
+	pkg_install() {
+		vmove usr/include/webkitgtk-4.0
+		vmove "usr/lib/pkgconfig/*-4.0.pc"
+		if [ "$build_option_gir" ]; then
+			vmove "usr/share/gir-1.0/*-4.0.gir"
+			vmove "usr/share/gtk-doc/html/*-4.0"
+		fi
+		vmove "usr/lib/*-4.0.so"
+	}
+}
+
+webkit2gtk41_package() {
+	depends="${_all_depends}"
+	short_desc="GTK+3 port of the WebKit2 browser engine (soup3)"
+	pkg_install() {
+		vmove usr/bin/WebKitWebDriver
+		vmove usr/libexec/webkit2gtk-4.1
+		vmove usr/lib/webkit2gtk-4.1
+		if [ "$build_option_gir" ]; then
+			vmove "usr/lib/girepository-1.0/*-4.1.typelib"
+		fi
+		vmove "usr/lib/*-4.1.so.*"
+	}
+}
+
+webkit2gtk41-devel_package() {
+	depends="gtk+3-devel libsoup3-devel ${sourcepkg}>=${version}_${revision}"
+	short_desc="GTK+3 port of the WebKit2 browser engine (soup3) - development files"
+	pkg_install() {
+		vmove usr/include/webkitgtk-4.1
+		vmove "usr/lib/pkgconfig/*-4.1.pc"
+		if [ "$build_option_gir" ]; then
+			vmove "usr/share/gir-1.0/*-4.1.gir"
+			vmove "usr/share/gtk-doc/html/*-4.1"
+		fi
+		vmove "usr/lib/*-4.1.so"
+	}
+}
+
+webkit2gtk50_package() {
```

Comment:
As above, this should probably be `libwebkit2gtk50` because its primary function is shipping a library (and its supporting content).

Reply:
All of their primary function is shipping a library, `WebDriver` is an language-agnostic api that allows you to control a browser, from say, a python script (i.e. for testing purposes):
https://developer.mozilla.org/en-US/docs/Web/WebDriver

So I don't see the need for the distinction. Feel free to disagree though.

Reply:
The point is that we tend to name library-specific subpackages with `lib` prefixes. If there were a compelling reason to split out the (v4.1 in my modified structure) library from the main package, that should be `libwebkit2gtk41` as well. However, because there's no value in making `webkit2gtk` an empty package that just pulls in the (default) library, just keep that as `webkit2gtk`.